app/model/block.py

```diff
@@ -3,7 +3,20 @@ def __init__(self):
         self.id = None
         self.title = None
         self.sections = []
+        self.container = None
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
+            section.container = self
+
+    def get_item_by_id(self, id):
```

Comment:
Since this function is identical in all instances below except for the parameter being iterated it is a prime candidate for refactoring into a utility function and only being implemented/tested once.